cmd/k8sagent/main_nix.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2012-2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/juju/clock"
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/featureflag"
+	"github.com/juju/loggo"
+	proxyutils "github.com/juju/proxy"
+
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/jujud/dumplogs"
+	"github.com/juju/juju/cmd/jujud/introspect"
+	"github.com/juju/juju/cmd/jujud/run"
+	cmdutil "github.com/juju/juju/cmd/jujud/util"
+	initcommand "github.com/juju/juju/cmd/k8sagent/init"
+	unitcommand "github.com/juju/juju/cmd/k8sagent/unit"
+	components "github.com/juju/juju/component/all"
+	"github.com/juju/juju/core/machinelock"
+	"github.com/juju/juju/juju/names"
+	"github.com/juju/juju/juju/osenv"
+	"github.com/juju/juju/utils/proxy"
+	"github.com/juju/juju/worker/logsender"
+)
+
+var logger = loggo.GetLogger("juju.cmd.k8sagent")
+
+func init() {
+	featureflag.SetFlagsFromEnvironment(osenv.JujuFeatureFlagEnvKey)
+
+	if err := components.RegisterForServer(); err != nil {
```

Comment:
These are primarily for payload and resource API server bits I think. By adding this in we are going to drag in all of the state package, and a bunch of other things we don't actually want.

We may need to readdress this at some stage if api client components are needed, but I'd rather just have them moved into the right place. This may be happening as a piece of work by other folks in the 2.9 branch anyway.

Reply:
Since the sub commands are empty, so removed this for now.
Might continue to resolve here in following PR when we introduced API accessing in those commands.

Reply:
Why is it still here then?